rotkehlchen/globaldb/handler.py

```diff
@@ -1202,22 +1202,22 @@ def get_historical_price(
         If no price can be found returns None
         """
         querystr = (
-            'SELECT from_asset, to_asset, source_type, timestamp, '
-            'price, MIN(ABS(timestamp - ?)) FROM price_history '
-            'WHERE from_asset=? AND to_asset=? '
-            'AND timestamp between ? AND ?'
+            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
+            'WHERE from_asset=? AND to_asset=? AND timestamp between ? AND ?'
         )
-        querylist = [timestamp, from_asset.identifier, to_asset.identifier, timestamp - max_seconds_distance, timestamp + max_seconds_distance]  # noqa: E501
+        querylist = [from_asset.identifier, to_asset.identifier, timestamp - max_seconds_distance, timestamp + max_seconds_distance]  # noqa: E501
         if source is not None:
             querystr += ' AND source_type=? '
             querylist.append(source.serialize_for_db())
 
+        # prioritize manual price if it exists and choose the timestamp closest to the target
+        querystr += ' ORDER BY CASE WHEN source_type=? THEN 0 ELSE 1 END, ABS(timestamp - ?)'
```

Comment:
this logic and the query logic is duplicated in this and the get_historical_prices. Can you maybe abstract what you can out?